application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,267 @@
+package org.togetherjava.tjbot.commands.bytecode;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.events.message.MessageDeleteEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.MessageUpdateEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.commands.MessageReceiver;
+import org.togetherjava.tjbot.imc.CompilationResult;
+import org.togetherjava.tjbot.imc.CompileInfo;
+import org.togetherjava.tjbot.imc.InMemoryCompiler;
+import org.togetherjava.tjbot.imc.JavacOption;
+import org.togetherjava.tjbot.javap.Javap;
+import org.togetherjava.tjbot.javap.JavapOption;
+
+import javax.tools.Diagnostic;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Bytecode command that uses the message context commands Example usage:
+ * <p>
+ *
+ * <pre>
+ * {@code
+ *     ```java
+ *     class Test {
+ *         public void foo() {
+ *             System.out.println("Hello World!");
+ *         }
+ *     }
+ *     ```
+ * }
+ * </pre>
+ *
+ * then apply the application
+ */
+public final class BytecodeCommand extends BotCommandAdapter
+        implements MessageContextCommand, MessageReceiver {
+    private static final String CODE_BLOCK_OPENING = "```rust\n";
+    private static final String CODE_BLOCK_CLOSING = "\n```";
+    private static final Pattern ANY_CHANNEL_NAME = Pattern.compile(".*");
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
+    private final Map<Long, List<Long>> userToBotMessages = new HashMap<>();
```

Comment:
* the naming on this is confusing, suggests that the key would be user ids, but its user message ids. maybe `originalToByteCodeMessages`
* this could explode RAM if abused. it should be limited in size (for example by using a Cafeeine cache, see other classes that used it)
* note that this silently kills the feature on bot restart. consider using the database if thats an issue for UX